tests/test_runner/test_debug_sql.py

```diff
@@ -0,0 +1,72 @@
+import unittest
+
+from django.db import connection
+from django.test import TestCase
+from django.test.runner import DiscoverRunner
+from django.utils import six
+
+from .models import Person
+
+
+@unittest.skipUnless(connection.vendor == 'sqlite', 'Only run on sqlite so we can check output SQL.')
+class TestDebugSQL(unittest.TestCase):
+
+    class PassingTest(TestCase):
+        def runTest(self):
+            Person.objects.filter(first_name='pass').count()
+
+    class FailingTest(TestCase):
+        def runTest(self):
+            Person.objects.filter(first_name='fail').count()
+            self.fail()
+
+    class ErrorTest(TestCase):
+        def runTest(self):
+            Person.objects.filter(first_name='error').count()
+            raise Exception
+
+    def test_output(self):
+        runner = DiscoverRunner(debug_sql=True, verbosity=0)
+        suite = runner.test_suite()
+        suite.addTest(self.FailingTest())
+        suite.addTest(self.ErrorTest())
+        suite.addTest(self.PassingTest())
+        old_config = runner.setup_databases()
+        stream = six.StringIO()
+        runner.test_runner(
+            verbosity=2,
+            stream=stream,
+        ).run(suite)
+        runner.teardown_databases(old_config)
+
+        stream.seek(0)
+        full_output = stream.read()
+        for output in self.expected_outputs:
+            self.assertIn(output, full_output)
+        for output in self.unexpected_outputs:
+            self.assertNotIn(output, full_output)
+
+    expected_outputs = [
+        'runTest (test_runner.test_debug_sql.FailingTest) ... FAIL',
+        'runTest (test_runner.test_debug_sql.ErrorTest) ... ERROR',
+        'runTest (test_runner.test_debug_sql.PassingTest) ... ok',
+    ]
+    if six.PY3:
+        expected_outputs += [
+            '''QUERY = 'SELECT COUNT(%s) AS "__count" FROM "test_runner_person" WHERE "test_runner_person"."first_name" = %s' - PARAMS = ('*', 'error');''',
```

Comment:
would be a bit cleaner to interpolate `django.test.utils.str_prefix` IMO. On the other hand, this will be easier to cleanup when we drop py2, so maybe it's fine. Could you limit the line lengths though?